lib/commands/project/project-command-base.ts

```diff
@@ -5,13 +5,11 @@ export class ProjectCommandBase {
 	constructor(protected $errors: IErrors,
 		protected $project: Project.IProject) { }
 
-	protected canExecuteCore(): IFuture<boolean> {
+	protected validateProjectData(): IFuture<boolean> {
```

Comment:
this method returns `IFuture<boolean>`, but you never return boolean. Consider returning `IFuture<void>`